core-plugins/src/main/java/co/cask/hydrator/plugin/batch/source/FileSourceConfig.java

```diff
@@ -149,28 +149,49 @@ protected void validate() {
       (format.equalsIgnoreCase("avro")) || format.equalsIgnoreCase("parquet"))) {
       throw new IllegalArgumentException("Format can only be 'text', 'avro' or 'parquet'");
     }
+    if (format.equalsIgnoreCase("text") && schema == null) {
```

Comment:
If schema is not specified, then we don't have to perform any of the checks that are after this. This felt like a cleaner option than adding everything else in another conditional block.